src/rospkg/os_detect.py

```diff
@@ -73,9 +73,11 @@ def read_issue(filename="/etc/issue"):
 
 def read_os_release(filename="/etc/os-release"):
     """
-    :returns: Dictonary of key value pairs from /etc/os-release, with quotes stripped from values
+    :returns: Dictonary of key value pairs from /etc/os-release or fallback to /usr/lib/os-release, with quotes stripped from values
     """
     release_info = {}
+    if not os.path.exists(filename):
+        filename = "/usr/lib/os-release"
```

Comment:
I find it weird that this fallback is being used even when a non-standard `filename` is passed to the function especially since the fallback can't be changed.

Reply:
`/etc/os-releaes`and `/usr/lib/os-release` are standard locations as said in
 https://www.freedesktop.org/software/systemd/man/os-release.html. 

When a non-standard `filename` is passed, then it must exist as the programmer is telling where the file live.

Reply:
> When a non-standard `filename` is passed, then it must exist as the programmer is telling where the file live.

That is true. I just wouldn't rely on it. Instead the default value of the argument could be a tuple instead and if the first item doesn't exist the next one is considered. That avoids falling back to the new location in case a custom `filename` is passed.

Reply:
In that case the method could be written as following:
* Change as empty the input parameter `filename`.
* If there is no input string path parameter then:
  + Check both locations _(as man os-release indicates)_.
* Else: 
  + Check only the input string `filename`.

I think in this particular method it is not necessary the usage of tuples, and doing it this way it is avoided falling back to the new location in case a custom `filename` is passed.